lib/model.js

```diff
@@ -1025,6 +1027,7 @@ module.exports = function(registry) {
           }
         });
       });
+      relation.visited = true;
```

Comment:
I don't think it's a good idea to change the global state of `relation`. I would rather have an `visited models` array local to this function.

Reply:
I'm not sure I understand what you mean; if I make it local scope, how will it survive between subsequent calls?

Or did you mean that the Model should keep a static array of visited relations?

Reply:
`var relation = this.relations[relationName];` is the metadata for the model and its state should not be mutated for remoting calculation unless it's one time charge for the whole life of the model def.

Reply:
To be honest, I'm not sure how to deal with this alternatively; I'm not familiar with this area of the codebase at a conceptual level, so if the level of effort to make the fix is similar and you know the area in which to do it, you might be better off making the PR instead.